api/src/nlp/services/nlp-sample.service.ts

```diff
@@ -235,11 +237,32 @@ export class NlpSampleService extends BaseService<
    *
    * @param language The language that has been deleted.
    */
-  @OnEvent('hook:language:delete')
-  async handleLanguageDelete(language: Language) {
+  @OnEvent('hook:language:preDelete')
+  async handleLanguageDelete(
+    _query: Query<
+      DeleteResult,
+      Document<Language, any, any>,
+      unknown,
+      Language,
+      'deleteOne' | 'deleteMany'
+    >,
+    criteria: TFilterQuery<Language>,
+  ) {
+    const deletedLanguages = await this.languageService.find(
+      criteria,
+      undefined,
+      {
+        id: 1,
+      },
+    );
+    const deletedLanguagesIds = deletedLanguages.map(
+      (deletedLanguage) => deletedLanguage.id,
+    );
     await this.updateMany(
       {
-        language: language.id,
+        language: {
+          $in: deletedLanguagesIds,
```

Comment:
What will be the behavior if deletedLanguagesIds is an empty array ? Ensuring the unit test covers this edge case can be beneficial.

Reply:
I think that we need to add an if condition to check if deletedLanguagesIds that has elements execute the the update